js/foundation.smoothScroll.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+!function($) {
+
+/**
+ * SmoothScroll module.
+ * @module foundation.smooth-scroll
+ */
+class SmoothScroll {
+    constructor(element, options) {
+        this.$element = element;
+        this.options = $.extend({}, SmoothScroll.defaults, this.$element.data(), options);
+
+        this._init();
+
+        Foundation.registerPlugin(this, 'SmoothScroll');
+    }
+
+    /**
+     * Initialize the SmoothScroll plugin 
+     */
+    _init() {
+        var id = this.$element[0].id || Foundation.GetYoDigits(6, 'smooth-scroll');
+        var _this = this;
+        this.$element.attr({
+            'id': id
+        });
+
+        this._events();
+    }
+
+    _events() {
+        var _this = this;
+
+        this.$element.on('click.zf.smoothScroll', 'a[href^="#"]', function(e) {
+            e.preventDefault();
+            var arrival   = this.getAttribute('href');
+            _this.scrollToLoc(arrival);
+        });
```

Comment:
It does not target a simple link, like:
```html
<a href="#target" data-smooth-scroll>Link</a>
```
We can move the code above to a public or private `onLinkClick` function, and then call:
```js
this.$element.on('click.zf.smoothScroll', _this.onLinkClick); // or onLinkClick
this.$element.on('click.zf.smoothScroll', 'a[href^="#"]', _this.onLinkClick); // or onLinkClick
```
... with a manual check of if `[href]` is set and begin with `#`.